core/src/main/java/io/questdb/griffin/SqlParser.java

```diff
@@ -786,13 +786,26 @@ private ExecutionModel parseCreateMatView(
         ));
 
         tok = tok(lexer, "'as' or 'with' or 'refresh'");
-        String baseTableName = null;
+        CharSequence baseTableName = null;
         if (isWithKeyword(tok)) {
             expectTok(lexer, "base");
-            baseTableName = Chars.toString(tok(lexer, "base table expected"));
+            mvOpBuilder.setBaseTableNamePosition(lexer.getPosition());
+            baseTableName = unquote(tok(lexer, "base table expected"));
+            if (isPublicKeyword(baseTableName)) {
```

Comment:
will this work with a source table called `public`? it's a valid table name after all. 

I have a stalled PR which adds some infrastructure to strip public schema safely: https://github.com/questdb/questdb/pull/5240/files#diff-51e5aeb7df979d7f612b9ad370be7a84a19bd88179c20a54747af64096d2f721R3211, perhaps I can extract this into its own PR or add it here.

Reply:
ok, i will move this PR into draft mode, and wait for yours to be merged.

Reply:
waiting for https://github.com/questdb/questdb/pull/5240